### ISSUE TITLE
ChannelPool swallows Exception leading to stuck rpc calls

### ISSUE THREAD

**Issue by doffid** [googleapis/sdk-platform-java#1332]
In the ChannelPool at `com.google.api.gax.grpc.ChannelPool.ReleasingClientCall#start` an Exception is caugth and swallowed.
This leads to a stuck grpc call if an exception is thrown (e.g. out of a ClientInterceptor) and does hide the causing exception.

The snippet in question:
```java
} catch (Exception e) {
  // In case start failed, make sure to release
  entry.release();
}
```
Maybe the simplest solution:
```java
} catch (RuntimeException e) {
  // In case start failed, make sure to release
  entry.release();
  throw e;
}
```

Thanks!

**Comment by JoeWang1127** [comment]
@blakeli0 Is this issue resolved by #1221?

**Comment by blakeli0** [comment]
> @blakeli0 Is this issue resolved by #1221?

Sorry for the late reply. Yes, it is resolved by #1221
